pkg/oauth/tokenrequest/request_token.go

```diff
@@ -284,6 +291,71 @@ func (o *RequestTokenOptions) RequestToken() (string, error) {
 	}
 }
 
+func (o *RequestTokenOptions) RequestTokenWithAuthorization(authorizeUrlHandler func(string) error, callbackPort int) (string, error) {
+	if o.OsinConfig == nil {
+		if err := o.SetDefaultOsinConfig(); err != nil {
+			return "", err
+		}
+	}
+
+	callbackServer, err := newCallbackServer(callbackPort)
+
+	listenAddr, ok := callbackServer.tcpListener.Addr().(*net.TCPAddr)
```

Comment:
store the serving URL in the object upon construction in `newCallbackServer()`, expose with a method

Reply:
See:
- https://github.com/openshift/library-go/pull/1531/files#diff-f1f7ca9dbfbf255b081f11049156e311316e2601aa5af53c7558f7f5e14a0d74R49-R53
- https://github.com/openshift/library-go/pull/1531/files#diff-f1f7ca9dbfbf255b081f11049156e311316e2601aa5af53c7558f7f5e14a0d74R65-R67
- https://github.com/openshift/library-go/pull/1531/files#diff-f30259f7a13841a226555f2801678f096da8bb80456cf414201b8cfd647da728R94